package/kedro_viz/api/rest/responses.py

```diff
@@ -333,7 +333,7 @@ def get_node_metadata_response(node_id: str):
         return TaskNodeMetadata(task_node=node)
 
     if isinstance(node, DataNode):
-        return DataNodeMetadata(data_node=node)
+        return DataNodeMetadata(data_node=node, is_preview_enabled_for_all_nodes=preview)
```

Comment:
as mentioned in a comment below, we can have `is_preview_enabled_for_all_nodes` or `is_preview_enabled` as a class attribute and should avoid passing this flag for all instances

Reply:
Done.